plugins/auth-backend-module-microsoft-provider/src/strategy.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PassportProfile } from '@backstage/plugin-auth-node';
+import { decodeJwt } from 'jose';
+import fetch from 'node-fetch';
+import { Strategy as MicrosoftStrategy } from 'passport-microsoft';
+
+export class ExtendedMicrosoftStrategy extends MicrosoftStrategy {
+  userProfile(
+    accessToken: string,
+    done: (err?: Error | null, profile?: PassportProfile) => void,
+  ): void {
+    if (this.skipUserProfile(accessToken)) {
+      done(null, undefined);
+      return;
+    }
+
+    super.userProfile(
+      accessToken,
+      async (err?: Error | null, profile?: PassportProfile) => {
+        if (profile) {
+          profile.photos =
+            profile.photos ?? (await this.getProfilePhotos(accessToken));
+        }
+        done(err, profile);
+      },
+    );
+  }
+
+  private skipUserProfile(accessToken: string): boolean {
+    const { aud, scp } = decodeJwt(accessToken);
+    const hasGraphReadScope =
+      aud === '00000003-0000-0000-c000-000000000000' &&
+      (scp as string)
+        .split(' ')
+        .map(s => s.toLowerCase())
+        .includes('user.read');
+    return !hasGraphReadScope;
+  }
+
+  private async getProfilePhotos(
+    accessToken: string,
+  ): Promise<Array<{ value: string }> | undefined> {
+    return this.getCurrentUserPhoto(accessToken, '96x96').then(photo =>
```

Comment:
Just noting that this does inflate catalog entity sizes significantly. Since it's jpeg it's probably not all too terrible, but for those who have tens or hundreds of thousands of org entities, it does turn out to be a fair amount of extra storage / bandwidth used overall. Might be worth an option in the future.

Reply:
Not added to the entities though, just the returned user profile from this provider

Reply:
oh right, of course